core/rawdb/freezer_table.go

```diff
@@ -153,8 +153,15 @@ func newTable(path string, name string, readMeter metrics.Meter, writeMeter metr
 		if err != nil {
 			return nil, err
 		}
-		// Will fail if the table is legacy(no metadata)
-		meta, err = openFreezerFileForReadOnly(filepath.Join(path, fmt.Sprintf("%s.meta", name)))
+		// TODO(rjl493456442) change it to read-only mode. Open the metadata file
+		// in rw mode. It's a temporary solution for now and should be changed
+		// whenever the tail deletion is actually used. The reason for this hack is
+		// the additional meta file for each freezer table is added in order to support
+		// tail deletion, but for most legacy nodes this file is missing. This check
+		// will suddenly break lots of database relevant commands. So the metadata file
+		// is always opened for mutation and nothing else will be written except
+		// the initialization.
+		meta, err = openFreezerFileForAppend(filepath.Join(path, fmt.Sprintf("%s.meta", name)))
 		if err != nil {
```

Comment:
Isn't it easier to just check `os.IsNotExist(err)` and ignore it here?

Reply:
In this case, the meta file handler is nil and we need to check if the handler is nil afterward.

Reply:
I think the hack is acceptable